src/customer-service/Actions/CreateIncidentAction.cs

```diff
@@ -0,0 +1,217 @@
+            Incident incident = new Incident();
+            incident.DateCreated = DateTime.Now;
+            while (incident.OrderId <= 0 || orderList.TrueForAll(o => incident.OrderId != orderID))
+            {
+                if (orderList.Find(o => o.OrderId == orderID) != null)
+                {
+                    break;
+                }
+
+                if (orderList.TrueForAll(o => incident.OrderId != orderID) || orderID == 0)
+                {
+                    Console.WriteLine("This is not a valid answer. Try again!");
+                    try
+                    {
+                        orderID = Convert.ToInt32(Console.ReadLine());
+
+                    }
+                    catch
+                    {
+                        Console.WriteLine("Please enter a number");
+                    }
+                }
+            }
+
+            incident.OrderId = orderID;
+            Console.WriteLine(@"
+                ====================================
+                BANGAZON INC CUSTOMER SERVICE PORTAL
+                ====================================");
+            Console.WriteLine("Choose incident type: ");
+            foreach(IncidentType incidentType in incidentTypeList)
+            {
+                Console.WriteLine($"{incidentType.IncidentTypeId} {incidentType.Label}");
+            }
+            Console.WriteLine("X.Exit");
+            int incidentTypeId = 0;
+            while (incidentTypeId == 0)
+            {
+                string incidentTypeIdAnswer = Console.ReadLine();
+                if (incidentTypeIdAnswer.ToLower() == "x")
+                {
+                    // to main menu
+                    return;
+                }
+                try
+                {
+                     incidentTypeId = Convert.ToInt32(incidentTypeIdAnswer);
+                }
+                catch
+                {
+                    Console.WriteLine("This is not a valid answer. Try again!");
+                }
+            }
+
+            while (incident.IncidentTypeId <= 0 || incidentTypeList.TrueForAll(o => incident.IncidentTypeId != incidentTypeId))
+            {
+                if (incidentTypeList.Find(it => it.IncidentTypeId == incidentTypeId) != null)
+                {
+                    break;
+                }
+
+                if (incidentTypeList.TrueForAll(o => incident.IncidentTypeId != incidentTypeId) || incidentTypeId == 0)
+                {
+                    Console.WriteLine("This is not a valid answer. Try again!");
+                    try
+                    {
+                        incidentTypeId = Convert.ToInt32(Console.ReadLine());
+
+                    }
+                    catch
+                    {
+                        Console.WriteLine("Please enter a number");
+                    }
+                }
+            }
+            incident.IncidentTypeId = incidentTypeId;
+

```

Comment:
Lose the code below here. Instead, set the `IncidentFactory.Instance.ActiveInstance` to the current instance and the `CustomerFactory.Instance.ActiveCustomer` to the current customer. Then call Dayne's `ShowSingleIncidentAction.ReadLine()`.